src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -539,13 +553,26 @@ int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
     int mpl_err = MPL_SUCCESS;
     gavl_tree_s *tree_ptr = (gavl_tree_s *) gavl_tree;
 
-    while (tree_ptr->root) {
-        int cmp_ret;
-        gavl_tree_node_s *dnode;
+    /* move all intersecting nodes with input buffer (addr, len) to remove_list */
+    gavl_tree_remove_nodes(tree_ptr, (uintptr_t) addr, len, INTERSECTION_SEARCH);
```

Comment:
"move all nodes intersecting with input buffer..."

Reply:
Besides, the second commit message reads confusing to me and has some grammar mistakes. How about:
```
gavl tree does not allow a node that is the subset of another. Therefore,
we remove all subset nodes of the input buffer during insertion. The removed 
nodes are maintained at a separate list in the tree in order to ensure correctness
if any outstanding operations are still using these nodes. The removed nodes and
buffer objects are freed only when MPL_gavl_tree_delete is called.
```